checker.cpp

```diff
@@ -1,22 +1,56 @@
-#include <assert.h>
-#include <iostream>
-using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7) == true);
-  assert(batteryIsOk(50, 85, 0) == false);
-}
+#include"checker.h"
+
+int main() {
+  cout<<"Battery Instance 1"<<endl;
+  checker checkerObj(new BatteryMgmtSystmTest(45, 20, 0.8));
+  assert(checkerObj.checkTemperature()==true);
+  assert(checkerObj.checkSoc()==true);
+  assert(checkerObj.checkChargeRate()==true);
+  assert(checkerObj.batteryIsOk()==true);
+  cout<<"Battery Instance 2"<<endl;
+  checkerObj.setParam(-1, -10, -0.9);
+  assert(checkerObj.checkTemperature()==false);
+  assert(checkerObj.checkSoc()==false);
+  assert(checkerObj.checkChargeRate()==false);
+  assert(checkerObj.batteryIsOk()==false);

```

Comment:
Does threshold monitoring get triggered in the batteryIsOk() call?

Reply:
Yes. All the three functions checkTemperature(), checkSoc(),checkChargeRate() gets called from batteryIsOk() call.

Reply:
Sorry, I wasn,t clear before. What I meant was, whether methods checkTempWarning() , checkSocWarning(), checkChargeRateWarning() are being used in batteryIsOk() call?

I could see that methods are present, have separate asserts (which is good approach), but could not trace it via batteryIsOk() method.